### DISCUSSION TITLE
Usage creates a classless div with 'data-overlay-container' attribute.

### DISCUSSION THREAD

**Discussion by MARCA-Charlie** [heroui-inc/heroui#804] [Feedback]
I am using in Next.JS, the div stops children growing to their parent's height and width. As the div is classless it is hard to target with styles to overcome this.

Please see screenshot for context.

![image](https://user-images.githubusercontent.com/89167030/194594372-4b369941-3461-4ae4-8ee0-39cc9c9c28d7.png)

**Comment by MARCA-Charlie** [comment]
e.g. the highlighted account container should grow to the height of website layout main div, but it cannot because of the next ui div in between.

**Comment by bsouthga** [comment]
You can select this element using an attribute selector (https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors)

```
[data-overlay-container=true] {
  height: 100%;
  width: 100%;
}
```

**Comment by Enalmada** [comment]
This still appears to be an issue in `0.0.0-dev-v2-20230719131057`.  It would be helpful for new users if this was fixed or a note added to the installation instructions.

**Reply by callum-gander** [reply]
Still an issue almost a year on, would be great to do the suggested above

**Reply by jrgarciadev** [reply]
https://github.com/nextui-org/nextui/pull/1513/files#diff-3980adffb068ab5925d2a958feb60542ba10b480bd38e9d1218abdd917b5d73a You can pass any props including style/className to the provider

This was tackle a couple week ago

**Reply by ckatala** [reply]
![Снимок экрана 2023-12-09 в 19 39 52](https://github.com/nextui-org/nextui/assets/46962268/5cd1ac24-4c23-4d6f-96d8-86a7c4dae68b)
Is this how it should be? For example, when I open popover, everything freezes terribly.

**Comment by Dilettante258** [comment]
The problem that has been bothering me for a long time😡

**Comment by tristanMatthias** [comment]
This is still an issue today. Would really appreciate a fix!

**Comment by eisa-rezaei** [comment]
data-overlay-container={true}
data-overlay-container={null}

getting hydration error for this
